src/pages/viewCohort/index.jsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect } from 'react';
+import { useState } from 'react';
+import { Link, useNavigate, useParams } from 'react-router-dom';
+import { Button } from '@mui/material';
+
+import Spinner from '../../components/Spinner';
+import client from '../../utils/client';
+import DEFAULTIMG from '../../assets/default.png';
+
+import './style.css';
+
+const ViewCohort = ({ setProfileView }) => {
+  const navigate = useNavigate();
+  const [cohort, setCohort] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const cohortId = parseInt(useParams().cohortId);
+  useEffect(() => {
+    client
```

Comment:
An if check in the useEffect or an early return if `cohortId` doesn't exist? Also is it possible for the param to not exist since the component is only rendered if the route `/cohort/:cohortId` matches?

Reply:
Yeah, an if in the `useEffect` or in the component (before the `useEffect`) would work. I don't think it'll be possible for `cohortId` to be non-existent, but it can easily be invalid (a string instead of a number). That being said, you don't know how the component will be used in the future. For instance, another dev could come along and try to use it incorrectly.

Reply:
The biggest reason though is to stop an error if a user enters the URL `/cohort/apple`, for example.